### ISSUE TITLE
Add a test using `alpaka_add_library`

### ISSUE THREAD

**Issue by bernhardmgruber** [alpaka-group/alpaka#2074]
> Do we have a test for an alpaka library by any chance? If not, we should open an issue.

We don't. We are also not using `alpaka_add_library` anywhere in this repository. Maybe PIConGPU or cupla make use of it?

_Originally posted by @j-stephan in https://github.com/alpaka-group/alpaka/issues/2072#issuecomment-1672825182_

**Comment by bernhardmgruber** [comment]
Thinking further about this, can we actually enable/disable accelerators for different libraries (but maybe sharing the same implementation)? Like:

```
enable CUDA
alpaka_add_library(cuda lib.cpp)
disable CUDA
enable OMP2
alpaka_add_library(omp lib.cpp) // same source
disable OMP2
add_executable(program main.cpp)
target_link_libraries(program cuda omp)
```

**Comment by j-stephan** [comment]
I don't think so (but maybe I'm wrong). But I had an idea for an alternative approach based on source file properties which might allow us to get rid of these special functions and just use the native `add_{library, executable}` directly.

**Comment by bernhardmgruber** [comment]
In any case, we should have a small example involving multiple libraries that show how a larger application with alpaka could be struktured. I only thought, since CMS also compiles a library with alpaka multiple times with different backends to dynamically load the right one from a driver executable, that this may be an interesting small test.

**Comment by SimeonEhrig** [comment]
Vikunja is testing `alpaka_add_library` in a special integration test: https://github.com/alpaka-group/vikunja/blob/9613b24ef5f9e0d66ef6d33e1781700c38d6f56c/ci/gitlab_scripts/integration_test/addsubdirectory.cmake#L47

In general, I implemented to different integration tests, to test the vikunja cmake with an external sample project. Maybe we should overtake the approach for alpaka.
